src/plugins/discover/public/components/discover_grid/discover_grid_flyout.test.tsx

```diff
@@ -71,30 +71,6 @@ describe('Discover flyout', function () {
     return { component, props };
   };
 
-  it('should be rendered correctly using an data view without timefield', async () => {
-    const { component, props } = mountComponent({});
-
-    const url = findTestSubject(component, 'docTableRowAction').prop('href');
-    expect(url).toMatchInlineSnapshot(`"/base/app/discover#/doc/the-data-view-id/i?id=1"`);
```

Comment:
The main advantage is that we are not losing history state `onClick` callback. We can use `href` from `getRedirectUrl` of locator, which will contain state, but it takes some time to load (see screenshot). I've suggested compromise solution in the last commit where we could copy redirect url or open in a new tab, but on just click it will use callbacks to faster navigation.

<img height="200" src="https://user-images.githubusercontent.com/39378793/197835056-fc76deea-7bf8-465e-aaf8-c6a535fd8ae8.jpeg" >

Reply:
didn't check the code now, as its late and I have to mentally prepare for 🎢 tomorrow  , but having a fallback solution while preferring the quicker way is a good solution in this case 👍

Reply:
interesting why it takes longer in the given example? Something worth to investigate

Reply:
Redirect url of locator always takes some time to open, it's used in `Explore data in Discover` button in Lens.

Reply:
Sure, done